remo/reports/api/views.py

```diff
@@ -19,3 +30,77 @@ def retrieve(self, request, pk):
         serializer = ActivitiesDetailedSerializer(report,
                                                   context={'request': request})
         return Response(serializer.data)
+
+
+class ActivitiesKPIFilter(django_filters.FilterSet):
+    """Filter for activities KPI endpoint."""
+    category = django_filters.CharFilter(name='functional_areas__name')
+    initiative = django_filters.CharFilter(name='campaign__name')
+
+    class Meta:
+        model = NGReport
+        fields = ['country']
+
+
+class ActivitiesKPIView(APIView):
+
+    def get(self, request):
+        """Returns serialized data for Activities KPI"""
+
+        queryset = NGReport.objects.filter(report_date__lte=now())
+        activities = ActivitiesKPIFilter(request.GET, queryset=queryset)
+        weeks = int(request.query_params.get('weeks', KPI_WEEKS))
+
+        # Total calculations
+        total = activities.qs.count()
+
+        # Quarter calculations
+        current_quarter = int(math.ceil(now().month/3.0))
+        current_quarter_start = datetime(now().year, current_quarter, 1)
+        quarter_total = activities.qs.filter(
+            report_date__gte=current_quarter_start).count()
+
+        # Total activities until start of quarter
+        quarter_s_total = activities.qs.filter(
```

Comment:
I'll have another take on this. The problem is that we end up with really large lines due to the calculations.